solr/core/src/java/org/apache/solr/search/facet/FacetRequest.java

```diff
@@ -229,19 +230,19 @@ public static void createJoinField(FacetRequest.Domain domain, Map<String,Object
        * Creates a Query that can be used to recompute the new "base" for this domain, relative to the
        * current base of the FacetContext.
        */
-      public Query createDomainQuery(FacetContext fcontext) throws IOException {
+      public Query createDomainQuery(FacetContext fcontext) {
         // NOTE: this code lives here, instead of in FacetProcessor.handleJoin, in order to minimize
         // the number of classes that have to know about the number of possible settings on the join
         // (ie: if we add a score mode, or some other modifier to how the joins are done)
 
-        final SolrConstantScoreQuery fromQuery = new SolrConstantScoreQuery(fcontext.base.getTopFilter());
+        final ConstantScoreQuery fromQuery = new ConstantScoreQuery(fcontext.base.makeQuery());
```

Comment:
Do we need to wrap this in a constant score query?

Reply:
Refactored it so `fromQuery` is not wrapped in a constant score query given we are using makeQuery which is a constant scoring query 👍